packages/sign_in_with_apple/lib/sign_in_with_apple.dart

```diff
@@ -47,7 +48,24 @@ class SignInWithApple {
     );
   }
 
-  static Future<bool> isAvailable() {
-    return _isAvailable ??= channel.invokeMethod<bool>('isAvailable');
+  /// Cached value whether or not sign-in with apple is available
+  ///
+  /// We cache, so we can return a [SynchronousFuture] in case this value has already been loaded
```

Comment:
@HenriBeck 

Even `SynchronousFuture` is not handled specially by the `FutureBuilder` (https://github.com/flutter/flutter/blob/0b8abb4724/packages/flutter/lib/src/widgets/async.dart#L566)

So I guess the only solution would be to not use the FB if we have a value.

Reply:
@tp It doesn't have to handle it. The `SynchronousFuture ` executes the `then` callback synchronously, which should result in the widget not needing a frame to resolve the future.

Reply:
You're right @HenriBeck. It will call `_subscribe` during `initState`, in which then `then` and hence `setState` are executed synchronously, so the data will be there for the first rendering.

https://github.com/flutter/flutter/blob/0b8abb4724/packages/flutter/lib/src/widgets/async.dart#L661